clippy_lints/src/loops/never_loop.rs

```diff
@@ -170,3 +198,27 @@ fn never_loop_expr_branch<'a, T: Iterator<Item = &'a Expr<'a>>>(e: &mut T, main_
     e.map(|e| never_loop_expr(e, main_loop_id))
         .fold(NeverLoopResult::AlwaysBreak, combine_branches)
 }
+
+fn for_to_if_let_sugg(
+    cx: &LateContext<'_>,
+    loop_expr: &Expr<'_>,
+    iterator: &Expr<'_>,
+    pat: &Pat<'_>,
+) -> Option<String> {
+    // Check whether the iterator expr needs parens to parse as `(iterator).into_iter()`.
+    // Ranges need special casing because they are converted to a struct in HIR.
+    let (open_paren, close_paren) = if iterator.precedence().order() < PREC_POSTFIX || is_range_literal(iterator) {
+        ("(", ")")
+    } else {
+        ("", "")
+    };
+    let iterator_snippet = snippet_opt(cx, iterator.span)?;
+    let pat_snippet = snippet(cx, pat.span, "_");
+
+    #[rustfmt::skip]
+    let sugg = format!(
+"if let Some({pat}) = {open_paren}{iterator}{close_paren}.into_iter().next() {{
+    /* ... */
+}}", pat = pat_snippet, open_paren = open_paren, iterator = iterator_snippet, close_paren = close_paren);
+    Some(reindent_multiline(sugg.into(), true, indent_of(cx, loop_expr.span)).into())
```

Comment:
Use `has_iter_method` to suggest `iter()` instead of `into_iter()` when applicable.